Drafts download and deployment docs for ELSER (#2371)
elastic/stack-docs: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc

:keywords: {ml-init}, {stack}, {nlp}, ELSER
:description: ELSER is a learned sparse ranking model trained by Elastic.

experimental[]

> Elastic Learned Sparse Encoder - or ELSER - is a text expansion model that uses 
> sparse vector representation and trained by Elastic . As a retrieval model, 

Reviewer: ```suggestion
Elastic Learned Sparse Encoder - or ELSER - is a representational model trained by Elastic that creates a
sparse vector representation  . As a retrieval model, 
```

Text expansion the _user friendly_ name we have chosen for the search query but it isn't a common term that people will recognise. I like @qherreros description of expanding text, maybe use the 'text expansion' after that

ELSER performs text-expansion for more relevant search results with expanding a 
search query to include terms that are not present in the query itself, but 
semantically highly relevant.

You can use ELSER for various NLP tasks, such as 
{ref}/semantic-search-elser.html[semantic search]. ELSER is an out-of-domain 
model which doesn't need to be fine-tuned on your data - it provides relevant 
search results out of the box.

// TO DO: Refer to this blog post, to learn more about ELSER.

[discrete]
[[elser-req]]
== Requirements

To use ELSER, you must have the {subscriptions}[appropriate subscription] level 
or the trial period activated.

[discrete]
[[download-elser]]
== Download ELSER

You can download ELSER either from the Trained Models page in {kib} or by using 

> the Dev Console.

Reviewer 2: Also from Enterprise Search - see https://github.com/elastic/kibana/pull/156080.

I am not sure how heavily we should push people towards the Enterprise Search app in the ML docs, but Enterprise Search is where we're planning to make the semantic search experience as friendly as possible, so we should certainly mention it here.

Author: I'm adding a section that covers this method, too.

:keywords: {ml-init}, {stack}, {nlp}, ELSER
:description: ELSER is a learned sparse ranking model trained by Elastic.

> Elastic Learned Sparse Encoder - or ELSER - is a large language model (LLM) 

Reviewer 2: Please add the tech preview banner to this page.

2. Click the Download model button under **Actions**. You can check the download 
status on the **Notifications** page.
+
--
[role="screenshot"]
image::images/ml-nlp-elser-download.png[alt="Downloading ELSER",align="center"]
--

Your ELSER deployment is ready to be used in 

[discrete]
[[dev-console]]
=== Using the Dev Console

1. In {kib}, navigate to the **Dev Console**.
2. Create the ELSER model configuration by running the following API call:

[source,console]
----------------------------------
PUT _ml/trained_models/.elser_model_1
{
  "input": {
	"field_names": ["text_field"]
  }
}
----------------------------------

The API call automatically initiates the model download if the model is not 
downloaded yet.

////
[discrete]
[[monitoring-progress]]
=== Monitoring the download progess
////

[discrete]
[[deploy-elser]]
== Deploy ELSER

After you downloaded ELSER, you can use {kib} to view and manage its deployment 
across your cluster under **{ml-app}** > **Trained Models**.

1. Start the deployment by clicking the Start deployment button.
2. Provide a deployment ID, select priority, and set the number of allocations 
and threads per allocation values.
+
--
[role="screenshot"]

> image::images/ml-nlp-elser-deployment.png[alt="Deploying ELSER",align="center"]

Author: ```suggestion
image::images/ml-nlp-deploy-elser.png[alt="Deploying ELSER",align="center"]
```

experimental[]

> Elastic Learned Sparse Encoder - or ELSER - is a representational model trained 

Reviewer: ```suggestion
Elastic Learned Sparse EncodeR - or ELSER - is a representational model trained 
```

For the R in ELSER. Is that correct @qherreros

Reviewer 3: Yes, true!

experimental[]

Elastic Learned Sparse Encoder - or ELSER - is a representational model trained 
by Elastic that creates a sparse vector representation of a text. As a retrieval 
model, ELSER performs text-expansion for more relevant search results with 
expanding a passage to a sparse representation of tokens that are carefully 
chosen to improve a semantically relevant retrieval.

You can use ELSER for various NLP tasks, such as 
{ref}/semantic-search-elser.html[semantic search]. ELSER is an out-of-domain 
model which doesn't need to be fine-tuned on your data - it provides relevant 
search results out of the box.

// TO DO: Refer to this blog post, to learn more about ELSER.

[discrete]
[[elser-req]]
== Requirements

To use ELSER, you must have the {subscriptions}[appropriate subscription] level 
or the trial period activated.

[discrete]
[[download-elser]]
== Download ELSER

> You can download ELSER either from the **{ml-app}** > **Trained Models** page or 
> by using the Dev Console.

Reviewer 2: ```suggestion
You can download ELSER either from the **{ml-app}** > **Trained Models** page, 
by using the Dev Console, or using {ents}.
```

Author: @droberts195 If I understand [this PR](https://github.com/elastic/kibana/pull/156080) correctly, you need to have ELSER downloaded in order to deploy the model through the Enterprise Search application. This is the reason why I didn't mention Enterprise Search here, only in the section about deploying the model.

Author: Okay, I modified in [94d47dd](https://github.com/elastic/stack-docs/pull/2371/commits/94d47dda64547413fa17e280efaa30988499e8cc).

Reviewer 4: > it's possible to both download and deploy ELSER just using the Enterprise Search app right?

Yes, correct.

. Repeat step 2 and step 3 on all master-eligible nodes.
. {ref}/restart-cluster.html#restart-cluster-rolling[Restart] the 
master-eligible nodes one by one.

[discrete]
[[further-readings]]

> == Further readings

Reviewer 2: ```suggestion
== Further reading
```
